lib/pacemaker/pcmk_sched_colocation.c

```diff
@@ -1409,14 +1428,18 @@ add_node_scores_matching_attr(GHashTable *nodes, const pe_resource_t *rsc,
  *
  * \param[in,out] rsc         Resource to check colocations for
  * \param[in]     log_id      Resource ID for logs (if NULL, use \p rsc ID)
- * \param[in,out] nodes       Nodes to update
+ * \param[in,out] nodes       Nodes to update (pass NULL as initial value to
+ *                            copy \p rsc's allowed nodes)
  * \param[in]     colocation  Original colocation constraint (used to get
  *                            configured primary resource's stickiness, and
- *                            to get colocation node attribute; pass NULL to
- *                            skip adding \p rsc's own node scores)
+ *                            to get colocation node attribute; if NULL,
+ *                            \p rsc's own node scores will not be added,
+ *                            and nodes must be NULL as well)
```

Comment:
> \p rsc's own node scores will not be added

Assuming `*nodes == NULL`, doesn't the below add `rsc`'s own node scores?
```
work = pcmk__copy_node_table(rsc->allowed_nodes);
```

> nodes must be NULL as well

May be worth clarifying `*nodes` (dereferenced)

Reply:
Updated comment